### PR TITLE
feat(searchbox): appbaseConfig prop for analytics

### PR DESCRIPTION
**Pull request by ShahAnuj2610**

### CODE REVIEW

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 1]
```diff
@@ -134,33 +135,9 @@ Example uses of searchbox UI:
...
-    			device: 'iphoneX',
-    		},
-    	}}
-    />
-    ```
-
-    > Note: **analyticsConfig** prop will only work when **analytics={true}**
+-   **appbaseConfig** `Object` [optional]
+    allows you to customize the analytics experience when appbase.io is used as a backend.
+    Read more about it over [here](/docs/reactivesearch/v3/advanced/analytics/#configure-the-analytics-experience).
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @ShahAnuj2610 We need to define the properties here, not all properties are present here.

File: content/docs/reactivesearch/searchbase/overview/apireference.md [hunk 2]
```diff
@@ -307,7 +308,9 @@ const searchbase = new Searchbase({
     index: "appname",
     url: "https://scalr.api.appbase.io",
     credentials: "abcdef123:abcdef12-ab12-ab12-ab12-abcdef123456",
-    analytics: true,
+    appbaseConfig: {
```
Comment (bietkul) [inline content/docs/reactivesearch/searchbase/overview/apireference.md:316]: @ShahAnuj2610 Use all options in the examples.

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 3]
```diff
@@ -41,6 +41,7 @@ Example uses of searchbox UI:
 ```js
 <SearchBox
 	app="good-books-ds"
+	appbaseConfig={{ recordAnalytics: true }}
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @ShahAnuj2610 Use all options.

File: content/docs/reactivesearch/vue-searchbox/apireference.md [hunk 4]
```diff
@@ -41,6 +41,9 @@ Example uses of searchbox UI:
 ```html
 <vue-searchBox
 	app="good-books-ds"
+	:appbaseConfig="{
```
Comment (bietkul) [inline content/docs/reactivesearch/vue-searchbox/apireference.md:44]: same here

File: content/docs/reactivesearch/vue-searchbox/apireference.md [hunk 5]
```diff
@@ -136,32 +139,9 @@ Example uses of searchbox UI:
...
-    			device: 'iphoneX',
-    		},
-    	}"
-    />
-    ```
-
-    > Note: **analyticsConfig** prop will only work when **analytics={true}**
+-   **appbaseConfig** `Object` [optional]
+    allows you to customize the analytics experience when appbase.io is used as a backend.
+    Read more about it over [here](/docs/reactivesearch/v3/advanced/analytics/#configure-the-analytics-experience).
```
Comment (bietkul) [inline content/docs/reactivesearch/vue-searchbox/apireference.md]: same here

**Review by bietkul — CHANGES_REQUESTED** [review]

**Review by siddharthlatest — APPROVED** [review]

**Review by siddharthlatest — APPROVED** [review]

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 6]
```diff
@@ -41,6 +41,15 @@ Example uses of searchbox UI:
 ```js
 <SearchBox
 	app="good-books-ds"
+	appbaseConfig={{
+        recordAnalytics: true,
+        enableQueryRules: true,
+        userId: '192.168.0.102',
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md:47]: @ShahAnuj2610 It shouldn't be the `IP`, it should be like `jon@appbase.io`.

File: content/docs/reactivesearch/searchbase/overview/apireference.md [hunk 7]
```diff
@@ -307,7 +313,15 @@ const searchbase = new Searchbase({
...
+        recordAnalytics: true,
+        enableQueryRules: true,
+        userId: '192.168.0.102',
```
Comment (bietkul) [inline content/docs/reactivesearch/searchbase/overview/apireference.md:319]: @ShahAnuj2610 Update

File: content/docs/reactivesearch/vue-searchbox/apireference.md [hunk 8]
```diff
@@ -41,6 +41,15 @@ Example uses of searchbox UI:
 ```html
 <vue-searchBox
 	app="good-books-ds"
+	:appbaseConfig="{
+        recordAnalytics: true,
+        enableQueryRules: true,
+        userId: '192.168.0.102',
```
Comment (bietkul) [inline content/docs/reactivesearch/vue-searchbox/apireference.md:47]: @ShahAnuj2610 Update

**Pull request merged**